### ISSUE TITLE
`apt new` is broken

### ISSUE THREAD

**Issue by maphew** [maphew/apt#29]
```
Package             Installed   Available
-----------------   ---------   ----------

TypeError: no_package() takes at least 2 arguments (0 given)
```

**Comment by amr66** [comment]
i saw a call to no_package in get_version, no parameters used there!
def get_version(packagename):
    if not dists[distname].has_key(packagename) \
       or not dists[distname][packagename].has_key(INSTALL):
        no_package()
        return (0, 0)
should be no_package(packagenane, distname), but aren't they global?

Dr. Andreas Müller -_- Universität Trier - FB VI - Kartographie, muellea@uni-trier.de
54286 Trier, Behringstr. 29, F 191, ++496512014576
________________________________
Von: matt wilkie [notifications@github.com]
Gesendet: Sonntag, 22. März 2015 21:43
An: maphew/apt
Betreff: [apt] `apt new` is broken (#29)


Package Installed Available ----------------- --------- ---------- TypeError: no_package() takes at least 2 arguments (0 given)

**Comment by maphew** [comment]
This reaches deep into the oldest code, well before my time, as well as solidly into some of my core changes. I made code easier to read and understand, but I made a fundamental mistake: *version* is now being read and reported straight from setup.ini dict and NOT from installed_db dict!

I have a fix queued for this to be committed shortly, but the logic error behind it is pernicious and is likely to cause troubles elsewhere, if it hasn't already. For example the archive file name and paths reported by `get_info()` and friends also generate their data from setup.ini and not from the actual local conditions. There needs to be a clear data model for local and remote.
